nim-mode.el

```diff
@@ -134,7 +134,11 @@
   (setq-local electric-indent-inhibit t)
   (setq-local electric-indent-chars '(?: ?\s))
   (when electric-indent-mode
-    (define-key nim-mode-map [remap delete-backward-char] 'nim-electric-backspace)))
+    (define-key nim-mode-map [remap delete-backward-char] 'nim-electric-backspace))
+
+
+  ;; Set `compile-command' to the correct initial value
+  (nim-compile--set-compile-command))
```

Comment:
if so, you can't put it on `nim-mode.el` as well? I'm not sure what you are trying to do. 
I think calling the function either on nim-mode-hook and file top-level isn't right way if users don't prepare nim-project yet when they open empty .nim file or if they did `(require 'nim-mode)` on not-nim-related buffer.